tests/lib/RequestParser.spec.js

```diff
@@ -0,0 +1,221 @@
+        path = "m/44'/242'/0'/6'";
+        expect(requestParser.parsePath(path, '5')).toEqual(path);
+    });
+
+    it('can parse Path Arrays', () => {
+        const requestParser = new RequestParser();
+
+        /** @type {any} */
+        let paths = undefined;
+        expect( () => { requestParser.parsePathsArray(paths, '1'); }).toThrow();
+        paths = 5;
+        expect( () => { requestParser.parsePathsArray(paths, '1'); }).toThrow();
+        paths = { something: 5 };
+        expect( () => { requestParser.parsePathsArray(paths, '1'); }).toThrow();
+        paths = [];
+        expect( () => { requestParser.parsePathsArray(paths, '1'); }).toThrow();
+        paths = ['something'];
+        expect( () => { requestParser.parsePathsArray(paths, '1'); }).toThrow();
+        paths = ["m/44'/242'/0'/6'","m/44'/242'/0'/6'"];
+        expect(requestParser.parsePathsArray(paths, '1')).toEqual(paths);
+    });
+
+    it('can parse label', () => {
+        const requestParser = new RequestParser();
+
+        /** @type {any} */
+        let label = undefined;
+        expect(requestParser.parseLabel(label)).toBe(undefined);
+        label = 5;
+        expect( () => { requestParser.parseLabel(label); }).toThrow();
+        label = '';
+        expect( requestParser.parseLabel(label)).toBe(undefined);
+        label = {};
+        expect( () => { requestParser.parseLabel(label); }).toThrow();
+        label = 'Ein Test';
+        expect(requestParser.parseLabel(label)).toEqual(label);
+    });
+
+    it('can parse KeyId to keyInfo', async () => {
+        const requestParser = new RequestParser();
+        let error;
+
+        /** @type {any} */
+        let keyId = undefined;
+        try {
+            const x = await requestParser.parseKeyId(keyId);
+        } catch(e) {
+            error = e;
+        }
+        expect(error).toEqual(new Errors.InvalidRequestError('keyId is required'));
+
+        keyId = {keyId: 75};
+        try {
+            const x = await requestParser.parseKeyId(keyId);
+        } catch(e) {
+            error = e;
+        }
+        expect(error).toEqual(new Errors.InvalidRequestError('keyId must be of type string'));
+
+        keyId = 'SuchMalNachDieserKeyId';
+        try {
+            const x = await requestParser.parseKeyId(keyId);
+        } catch(e) {
+            error = e;
+        }
+        expect(error).toEqual(new Errors.KeyNotFoundError());
+
+        keyId = 5;
+        try {
+            const x = await requestParser.parseKeyId(keyId);
+        } catch(e) {
+            error = e;
+        }
+        expect(error).toEqual(new Errors.InvalidRequestError('keyId must be of type string'));
+
+
+        keyId = '2ec615522906'; // dummy data
+        let parsedKeyInfo = await requestParser.parseKeyId(keyId);
+        expect(new KeyInfo(
+            '2ec615522906',
+            Key.Type.LEGACY,
+            true,
+            false,
+        )).toEqual(parsedKeyInfo);
```

Comment:
The expect should be the other way around here:

expect _test result_ to be _known data_.